File: manual_fixtures/integration_guide/web_server.js
var http = require('http');

var async = require('async');
var Client = require('service-registry-client/lib/client').Client

var username = ''; // your username here
var key = ''; // your API key here

function main() {
  var client = new Client(username, key, 'us', null);

  async.waterfall([
    function startHttpServer(callback) {
      var server = http.createServer(function (req, res) {
        res.writeHead(200, {'Content-Type': 'text/plain'});
        res.end('Hello, world!');
      };

      server.listen(8080, '127.0.0.1', callback);
    },

    function createService(callback) {
      var payload = {'tags': ['api']};

      client.services.register('web-service-api0', 30, payload, function(err, resp, hb) {
        callback(null, hb);
      });
    },

    function startHeartbeating(hb, callback) {
      hb.start();
      callback();
    }
  ],

  function(err) {
    if (err) {
      console.log('An error has occurred.');
      console.log(err);
    }
  });
}

main();

File: manual_fixtures/integration_guide/web_server.py
from twisted.web import server, resource
from twisted.internet import reactor

from txServiceRegistry import Client

RACKSPACE_USERNAME = '' # your Rackspace username here
RACKSPACE_KEY = '' # your Rackspace API key here

client = Client(RACKSPACE_USERNAME,
                RACKSPACE_KEY,
                'us')

class Simple(resource.Resource):
    isLeaf = True
    def render_GET(self, request):
        return "<html>Hello, world!</html>"


def cbService(result):
    global heartBeater
    heartBeater = result[1]
    heartBeater.start()

payload = {'tags': ['api']}

d = client.services.register('web-service-api0', 30, payload)
d.addCallback(cbService)

site = server.Site(Simple())
reactor.listenTCP(8080, site)
reactor.run()
